internal/usecase/interactor/scene.go

```diff
@@ -193,12 +193,17 @@ func (i *Scene) AddWidget(ctx context.Context, id id.SceneID, pid id.PluginID, e
 		return nil, nil, err
 	}
 
-	widget, err = scene.NewWidget(nil, pid, eid, property.ID(), true)
+	widgetLayout := extension.WidgetLayout()
+
+	widget, err = scene.NewWidget(nil, pid, eid, property.ID(), true, &scene.WidgetLayout{Extendable: widgetLayout.Extendable, Extended: widgetLayout.Extended, Floating: widgetLayout.Floating, DefaultLocation: widgetLayout.DefaultLocation})
 	if err != nil {
 		return nil, nil, err
 	}
 
 	s.WidgetSystem().Add(widget)
+	if !widget.WidgetLayout().Floating {
+		s.WidgetAlignSystem().Add(widget.ID().Ref(), widget.WidgetLayout().DefaultLocation)
+	}
```

Comment:
This comment is more developmental advice. If you have time, try it.

Currently, the scene has a WidgetSystem, which is a map of widget IDs and properties, and a WidgetAlignSystem. This means that we now have to describe in the use case layer the changes we want to happen in the WidgetAlignSystem when we add a widget to the WidgetSystem. This requires such "special" knowledge from the implementor, that when you forget to implement it, you get a bug.

Therefore, incorporating the WidgetAlignSystem in the WidgetSystem makes the code safer and less bug-prone, since this process can be hidden in the domain layer. The use case layer just needs to call widgetSystem.Add and everything is done.

Reply:
Since we are a little hurting for time right now, I am thinking this could be a good refactor task for me later in the month or next month.